### PR TITLE
COMMONS-487: Add owasp-java-html-sanitizer library to encode data in …

### PR DESCRIPTION
**Pull request by kmenzli**
…presentation layer

### CODE REVIEW

File: wiki-service/src/main/java/org/exoplatform/wiki/service/search/SearchResult.java [hunk 1]
```diff
@@ -76,7 +77,12 @@ public void setExcerpt(String text) {
   }
 
   public String getExcerpt() {
-    return Utils.sanitize(excerpt);
+    try {
+      return HTMLSanitizer.sanitize(excerpt);
+    } catch (Exception E) {
+
```
Comment (thomasdelhomenie) [inline wiki-service/src/main/java/org/exoplatform/wiki/service/search/SearchResult.java]: No empty catch block. You should add an error log here instead.

File: wiki-service/src/main/java/org/exoplatform/wiki/service/search/SearchResult.java [hunk 2]
```diff
@@ -76,7 +77,12 @@ public void setExcerpt(String text) {
   }
 
   public String getExcerpt() {
-    return Utils.sanitize(excerpt);
+    try {
+      return HTMLSanitizer.sanitize(excerpt);
+    } catch (Exception E) {
```
Comment (thomasdelhomenie) [inline wiki-service/src/main/java/org/exoplatform/wiki/service/search/SearchResult.java]: Variables must start by a lower case.

Comment (kmenzli) [inline wiki-service/src/main/java/org/exoplatform/wiki/service/search/SearchResult.java]: Which variables?

Comment (thomasdelhomenie) [inline wiki-service/src/main/java/org/exoplatform/wiki/service/search/SearchResult.java]: Exception E -> Exception e

File: wiki-service/src/main/java/org/exoplatform/wiki/service/search/SearchResult.java [hunk 3]
```diff
@@ -76,7 +80,14 @@ public void setExcerpt(String text) {
   }
 
   public String getExcerpt() {
-    return Utils.sanitize(excerpt);
+    try {
+      return HTMLSanitizer.sanitize(excerpt);
+    } catch (Exception e) {
+
+      log.error("Fail to sanitize input ["+excerpt+"]",e.getLocalizedMessage(),e);
```
Comment (thomasdelhomenie) [inline wiki-service/src/main/java/org/exoplatform/wiki/service/search/SearchResult.java]: You use the paramterized version of the error() method but there is no parameter in your message ( {} ).
It should be something like that instead :

`log.error("Fail to sanitize input [{}] : {}", excerpt, e.getMessage(), e);`

(I generally do not use getLocalizedMessage method, we use getMessage everywhere, and it is harder for support to read logs if there are non-english messages)

Comment (kmenzli) [inline wiki-service/src/main/java/org/exoplatform/wiki/service/search/SearchResult.java]: > (I generally do not use getLocalizedMessage method, we use getMessage everywhere, and it is harder for support to read logs if there are non-english messages)
You are right, we don't need support l10n/i18n here. Initially , I thought that we should consider contributors and customer developers working on the top of eXo, but from a product standpoint error message is not for the end customer but for the support engineer/development engineer.

**Pull request closed without merging**